### ISSUE TITLE
Create an empty column using None 

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#191]
## Feature Reqeust
In pandas, creating a column with the value None, creates an empty series instance. 

In pygdf doing the following:
```
import numpy as np
import pygdf
from pygdf.dataframe import DataFrame

gdf = DataFrame()
gdf['f0'] = None
print(gdf['f0'])
```
Gives the following output instead of an empty series:
```
      
0 
```

Executing it for multiple columns results in an error:
```
import numpy as np
import pygdf
from pygdf.dataframe import DataFrame

gdf = DataFrame()
gdf['f0'] = None
gdf['f1'] = None
```
Output:
```
KeyError                                  Traceback (most recent call last)
~/Downloads/ENTER/lib/python3.6/site-packages/numba/numpy_support.py in from_dtype(dtype)
     91         try:
---> 92             return FROM_DTYPE[dtype]
     93         except KeyError:

KeyError: dtype('O')

During handling of the above exception, another exception occurred:

NotImplementedError                       Traceback (most recent call last)
<ipython-input-47-3e56cd28e7dd> in <module>()
      5 gdf = DataFrame()
      6 gdf['f0'] = None
----> 7 gdf['f1'] = None

~/Downloads/ENTER/lib/python3.6/site-packages/pygdf/dataframe.py in __setitem__(self, name, col)
    182             self._cols[name] = self._prepare_series_for_add(col)
    183         else:
--> 184             self.add_column(name, col)
    185 
    186     def __delitem__(self, name):

~/Downloads/ENTER/lib/python3.6/site-packages/pygdf/dataframe.py in add_column(self, name, data, forceindex)
    393             raise NameError('duplicated column name {!r}'.format(name))
    394 
--> 395         series = self._prepare_series_for_add(data, forceindex=forceindex)
    396         self._cols[name] = series
    397 

~/Downloads/ENTER/lib/python3.6/site-packages/pygdf/dataframe.py in _prepare_series_for_add(self, col, forceindex)
    368         The prepared Series object.
    369         """
--> 370         col = self._sanitize_columns(col)
    371         empty_index = isinstance(self._index, EmptyIndex)
    372         series = Series(col)

~/Downloads/ENTER/lib/python3.6/site-packages/pygdf/dataframe.py in _sanitize_columns(self, col)
    337             self._size = len(series)
    338 
--> 339         col = self._sanitize_values(col)
    340         return col
    341 

~/Downloads/ENTER/lib/python3.6/site-packages/pygdf/dataframe.py in _sanitize_values(self, col)
    350         if len(self) > 0 and len(series) == 1 and not VALID:
    351             arr = cuda.device_array(shape=len(index), dtype=series.dtype)
--> 352             cudautils.gpu_fill_value.forall(arr.size)(arr, col)
    353             return Series(arr)
    354         elif len(self) > 0 and sind != index:

~/Downloads/ENTER/lib/python3.6/site-packages/numba/cuda/compiler.py in __call__(self, *args)
    247     def __call__(self, *args):
    248         if isinstance(self.kernel, AutoJitCUDAKernel):
--> 249             kernel = self.kernel.specialize(*args)
    250         else:
    251             kernel = self.kernel

~/Downloads/ENTER/lib/python3.6/site-packages/numba/cuda/compiler.py in specialize(self, *args)
    710         '''
    711         argtypes = tuple(
--> 712             [self.typingctx.resolve_argument_type(a) for a in args])
    713         kernel = self.compile(argtypes)
    714         return kernel

~/Downloads/ENTER/lib/python3.6/site-packages/numba/cuda/compiler.py in <listcomp>(.0)
    710         '''
    711         argtypes = tuple(
--> 712             [self.typingctx.resolve_argument_type(a) for a in args])
    713         kernel = self.compile(argtypes)
    714         return kernel

~/Downloads/ENTER/lib/python3.6/site-packages/numba/typing/context.py in resolve_argument_type(self, val)
    288         ValueError is raised for unsupported types.
    289         """
--> 290         return typeof(val, Purpose.argument)
    291 
    292     def resolve_value_type(self, val):

~/Downloads/ENTER/lib/python3.6/site-packages/numba/typing/typeof.py in typeof(val, purpose)
     27     # Note the behaviour for Purpose.argument must match _typeof.c.
     28     c = _TypeofContext(purpose)
---> 29     ty = typeof_impl(val, c)
     30     if ty is None:
     31         msg = errors.termcolor.errmsg(

~/Downloads/ENTER/lib/python3.6/functools.py in wrapper(*args, **kw)
    801 
    802     def wrapper(*args, **kw):
--> 803         return dispatch(args[0].__class__)(*args, **kw)
    804 
    805     registry[object] = func

~/Downloads/ENTER/lib/python3.6/site-packages/numba/typing/typeof.py in typeof_impl(val, c)
     53             return types.ffi
     54 
---> 55     return getattr(val, "_numba_type_", None)
     56 
     57 

~/Downloads/ENTER/lib/python3.6/site-packages/numba/cuda/cudadrv/devicearray.py in _numba_type_(self)
    145         represents this object.
    146         """
--> 147         dtype = numpy_support.from_dtype(self.dtype)
    148         return types.Array(dtype, self.ndim, 'A')
    149 

~/Downloads/ENTER/lib/python3.6/site-packages/numba/numpy_support.py in from_dtype(dtype)
     99                 subtype = from_dtype(dtype.subdtype[0])
    100                 return types.NestedArray(subtype, dtype.shape)
--> 101             raise NotImplementedError(dtype)
    102     else:
    103         return from_struct_dtype(dtype)

NotImplementedError: object
```
----

Current workaround includes creating empty lists instead of using `None`:
```
import numpy as np
import pygdf
from pygdf.dataframe import DataFrame

gdf = DataFrame()
gdf['f0'] = []
gdf['f1'] = []
```

**Comment by andersy005** [comment]
@ayushdg, thanks for opening this issue. I have a quick question though.. 
What version of pygdf are you on? Are you using the latest version of PyGDF from conda or built from master? I am unable to reproduce the issue. 

I am on 
```python
In [12]: pygdf.__version__
Out[12]: '0.1.0a2+293.g69f8656'
```


Instead of getting `0` for `print(gdf['f0'])`, I am getting  `AttributeError: 'NoneType' object has no attribute 'view'`

```python
In [11]: import numpy as np
    ...: import pygdf
    ...: from pygdf.dataframe import DataFrame
    ...:
    ...: gdf = DataFrame()
    ...: gdf['f0'] = None
    ...: print(gdf['f0'])
    ...:
    ...:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-9bceef0e732b> in <module>()
      5 gdf = DataFrame()
      6 gdf['f0'] = None
----> 7 print(gdf['f0'])

~/work/quansight/nvcollab/pygdf/pygdf/series.py in __str__(self)
    255
    256     def __str__(self):
--> 257         return self.to_string(nrows=10)
    258
    259     def __repr__(self):

~/work/quansight/nvcollab/pygdf/pygdf/series.py in to_string(self, nrows)
    249
    250         # Prepare cells
--> 251         cols = OrderedDict([('', self.values_to_string(nrows=nrows))])
    252         # Format into a table
    253         return formatting.format(index=self.index,

~/work/quansight/nvcollab/pygdf/pygdf/series.py in values_to_string(self, nrows)
    220         """
    221         values = self[:nrows]
--> 222         out = ['' if v is None else str(v) for v in values]
    223         return out
    224

~/work/quansight/nvcollab/pygdf/pygdf/series.py in <listcomp>(.0)
    220         """
    221         values = self[:nrows]
--> 222         out = ['' if v is None else str(v) for v in values]
    223         return out
    224

~/work/quansight/nvcollab/pygdf/pygdf/series.py in __getitem__(self, arg)
    176         elif isinstance(arg, Number):
    177             # The following triggers a IndexError if out-of-bound
--> 178             return self._column.element_indexing(arg)
    179         else:
    180             raise NotImplementedError(type(arg))

~/work/quansight/nvcollab/pygdf/pygdf/column.py in element_indexing(self, index)
    313         ``IndexError`` if out-of-bound
    314         """
--> 315         val = self.data[index]  # this can raise IndexError
    316         valid = (cudautils.mask_get.py_func(self.nullmask, index)
    317                  if self.has_null_mask else True)

~/work/quansight/nvcollab/pygdf/pygdf/buffer.py in __getitem__(self, arg)
    137             # cuda dev arrays, so the type of the cuda dev array is
    138             # an i64, and we view it as the dtype on the buffer
--> 139             return self.mem[arg].view(self.dtype)
    140         else:
    141             raise NotImplementedError(type(arg))

AttributeError: 'NoneType' object has no attribute 'view'
```

**Comment by ayushdg** [comment]
@andersy005, thank you for the response. I am on the latest version now and I get the same issue `AttributeError: 'NoneType' object has no attribute 'view'`. I might have been on an older version when I created the issue. Based on the results it seems like pygdf does not create an empty series on `None` assignment like pandas.

**Comment by randerzander** [comment]
Closed by #603
